webui/react/src/pages/TrialDetails.tsx

```diff
@@ -15,23 +15,68 @@ import usePolling from 'hooks/usePolling';
 import useRestApi from 'hooks/useRestApi';
 import TrialActions, { Action as TrialAction } from 'pages/TrialDetails/TrialActions';
 import TrialInfoBox from 'pages/TrialDetails/TrialInfoBox';
+import { routeAll } from 'routes';
+import { forkExperiment } from 'services/api';
 import { getExperimentDetails, getTrialDetails, isNotFound } from 'services/api';
 import { TrialDetailsParams } from 'services/types';
-import { ExperimentDetails, TrialDetails } from 'types';
+import { ExperimentDetails, RawJson, TrialDetails } from 'types';
 import { clone } from 'utils/data';
 import { trialHParamsToExperimentHParams } from 'utils/types';
 
 interface Params {
   trialId: string;
 }
 
+const getTrialLength = (experimentConfig: RawJson): [string, number] => {
+  const maxLength = experimentConfig.searcher.max_length;
+  if (!maxLength) {
+    // provide backward compat for max_steps
+    let maxSteps = experimentConfig.searcher.max_steps;
+    maxSteps = parseInt(maxSteps) || 100;
+    const batchesPerStep = experimentConfig.batches_per_step || 100;
+    return [ 'batches', maxSteps * batchesPerStep ];
+  }
+  return Object.entries(maxLength)[0] as [string, number];
+};
+
+const setTrialLength = (experimentConfig: RawJson, length: number): void => {
```

Comment:
**suggestion:** how about returning new `RawJson` (cloned) instead of mutating the exiting one?
Something like:
```
const setTrialLength = (experimentConfig: RawJson, length: number): RawJson => {
  const newConfig = clone(experimentConfig);
  ... do stuff to clone
  return newConfig;
};
```

Reply:
in the current usecase (and some future ones) the pattern can provide an already cloned version. I didn't clone here to avoided double cloning. I'm thinking the use can always provide a cloned object if they want to? or if the cloning isn't expensive we can always do it.

Reply:
ah if it's already cloned, you're right, no need to do it again.